<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>ProFAST: two DLPFC sections • ProFAST</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="ProFAST: two DLPFC sections">
<meta property="og:description" content="ProFAST">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ProFAST</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ProFASTdlpfc2.html">ProFAST: two DLPFC sections</a>
    </li>
    <li>
      <a href="../articles/ProFASTsimu.html">ProFAST: simulation</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/feiyoung/ProFAST/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>ProFAST: two DLPFC sections</h1>
                        <h4 data-toc-skip class="author">Wei Liu</h4>
            
            <h4 data-toc-skip class="date">2023-06-04</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/feiyoung/ProFAST/blob/HEAD/vignettes/ProFASTdlpfc2.Rmd" class="external-link"><code>vignettes/ProFASTdlpfc2.Rmd</code></a></small>
      <div class="hidden name"><code>ProFASTdlpfc2.Rmd</code></div>

    </div>

    
    
<p>This vignette introduces the ProFAST workflow for the analysis of two humn dorsolateral prefrontal cortex (DLPFC) spatial transcriptomics dataset. ProFAST workflow is based on the <code>PRECASTObj</code> object created in the <code>PRECAST</code> R package and the workflow of ProFAST is similar to that of PRECAST; see the vignette (<a href="https://feiyoung.github.io/PRECAST/articles/PRECAST.BreastCancer.html" class="external-link uri">https://feiyoung.github.io/PRECAST/articles/PRECAST.BreastCancer.html</a>) for the workflow of PRECAST. In this vignette, the workflow of ProFAST consists of three steps</p>
<ul>
<li>Independent preprocessing and model setting</li>
<li>Spatial dimension reduction using ProFAST</li>
<li>Downstream analysis (i.e. , embedding alignment and spatial clustering using iSC-MEB, visualization of clusters and embeddings, remove unwanted variation, combined differential expression analysis)</li>
<li>Except for the above downstream analyses, cell-cell interaction analysis and trajectory inference can also be performed on the basis of the embeddings obtained by ProFAST.</li>
</ul>
<p>We demonstrate the use of ProFAST to two DLPFC Visium data that are <a href="https://github.com/feiyoung/ProFAST/tree/main/vignettes_data" class="external-link">here</a>, which can be downloaded to the current working path by the following command:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">githubURL</span> <span class="op">&lt;-</span> <span class="st">"https://github.com/feiyoung/ProFAST/blob/main/vignettes_data/seulist2_ID9_10.RDS?raw=true"</span>
<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">githubURL</span>,<span class="st">"seulist2_ID9_10.RDS"</span>,mode<span class="op">=</span><span class="st">'wb'</span><span class="op">)</span></code></pre></div>
<p>Then load to R</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dlpfc2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"./seulist2_ID9_10.RDS"</span><span class="op">)</span></code></pre></div>
<p>The package can be loaded with the command:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/feiyoung/ProFAST" class="external-link">ProFAST</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/feiyoung/PRECAST" class="external-link">PRECAST</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: parallel</span>
<span class="co">#&gt; Loading required package: gtools</span>
<span class="co">#&gt; PRECAST :  An efficient data integration method is provided for multiple spatial transcriptomics data with non-cluster-relevant effects such as the complex batch effects. It unifies spatial factor analysis simultaneously with spatial clustering and embedding alignment, requiring only partially shared cell/domain clusters across datasets. More details can be referred to Wei Liu, et al. (2023) &lt;doi:10.1038/s41467-023-35947-w&gt;.   Check out our Package website (https://feiyoung.github.io/PRECAST/index.html) for a more complete description of the methods and analyses</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: package 'Seurat' was built under R version 4.1.3</span>
<span class="co">#&gt; Attaching SeuratObject</span>
<span class="co">#&gt; Attaching sp</span></code></pre></div>
<div class="section level2">
<h2 id="view-the-dlpfc-data">View the DLPFC data<a class="anchor" aria-label="anchor" href="#view-the-dlpfc-data"></a>
</h2>
<p>First, we view the the two spatial transcriptomics data with ST platform. There are ~15000 genes for each data batch and ~7000 spots in total.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dlpfc2</span> <span class="co">## a list including three Seurat object with default assay: RNA</span>
<span class="co">#&gt; [[1]]</span>
<span class="co">#&gt; An object of class Seurat </span>
<span class="co">#&gt; 15027 features across 3639 samples within 1 assay </span>
<span class="co">#&gt; Active assay: RNA (15027 features, 2000 variable features)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]]</span>
<span class="co">#&gt; An object of class Seurat </span>
<span class="co">#&gt; 15809 features across 3673 samples within 1 assay </span>
<span class="co">#&gt; Active assay: RNA (15809 features, 2000 variable features)</span></code></pre></div>
<p>Check the content in <code>dlpfc2</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">dlpfc2</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p>We observed the genes are Ensembl IDs. In the following, we will transfer the Ensembl IDs to gene symbols for matching the housekeeping genes in the downstream analysis for removing the unwanted variations.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">dlpfc2</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="create-a-precastobject-object">Create a PRECASTObject object<a class="anchor" aria-label="anchor" href="#create-a-precastobject-object"></a>
</h2>
<p>We show how to create a PRECASTObject object step by step. First, we create a Seurat list object using the count matrix and meta data of each data batch.</p>
<ul>
<li>Note: the spatial coordinates must be contained in the meta data and named as <code>row</code> and <code>col</code>, which benefits the identification of spaital coordinates by ProFAST</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Get the gene-by-spot read count matrices</span>
<span class="va">countList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">dlpfc2</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">[[</span><span class="st">"RNA"</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">counts</span><span class="op">)</span>
<span class="va">M</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">countList</span><span class="op">)</span>
<span class="co">### transfer the Ensembl ID to symbol</span>
<span class="kw">for</span><span class="op">(</span><span class="va">r</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">M</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">countList</span><span class="op">[[</span><span class="va">r</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/transferGeneNames.html">transferGeneNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">countList</span><span class="op">[[</span><span class="va">r</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, now_name <span class="op">=</span> <span class="st">"ensembl"</span>,
                                                 to_name<span class="op">=</span><span class="st">"symbol"</span>,
                                                 species<span class="op">=</span><span class="st">"Human"</span>, Method<span class="op">=</span><span class="st">'eg.db'</span><span class="op">)</span>
<span class="op">}</span>
<span class="co">#&gt; Loading required package: org.Hs.eg.db</span>
<span class="co">#&gt; Loading required package: AnnotationDbi</span>
<span class="co">#&gt; Loading required package: stats4</span>
<span class="co">#&gt; Loading required package: BiocGenerics</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'BiocGenerics'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     IQR, mad, sd, var, xtabs</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
<span class="co">#&gt;     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
<span class="co">#&gt;     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
<span class="co">#&gt;     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
<span class="co">#&gt;     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
<span class="co">#&gt;     union, unique, unsplit, which.max, which.min</span>
<span class="co">#&gt; Loading required package: Biobase</span>
<span class="co">#&gt; Welcome to Bioconductor</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     Vignettes contain introductory material; view with</span>
<span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="co">#&gt; Loading required package: IRanges</span>
<span class="co">#&gt; Loading required package: S4Vectors</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'S4Vectors'</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand.grid, I, unname</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'IRanges'</span>
<span class="co">#&gt; The following object is masked from 'package:sp':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     %over%</span>
<span class="co">#&gt; The following object is masked from 'package:grDevices':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     windows</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span>
<span class="co">#&gt; 'select()' returned 1:many mapping between keys and columns</span>

<span class="co">## Check the spatial coordinates: Yes, they are named as "row" and "col"!</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">dlpfc2</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt;                       orig.ident nCount_RNA nFeature_RNA            barcode</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1 SeuratProject       8436         3564 AAACAAGTATCTCCCA-1</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1 SeuratProject       1662         1145 AAACAATCTACTAGCA-1</span>
<span class="co">#&gt; AAACACCAATAACTGC-1 SeuratProject       3758         1949 AAACACCAATAACTGC-1</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1 SeuratProject       5418         2409 AAACAGAGCGACTCCT-1</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1 SeuratProject       4262         2248 AAACAGCTTTCAGAAG-1</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1 SeuratProject       3984         2159 AAACAGGGTCTATATT-1</span>
<span class="co">#&gt;                    sample_name tissue row col imagerow imagecol Cluster height</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1      151673      1  50 102 381.0981 440.6391       7    600</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1      151673      1   3  43 126.3276 259.6310       4    600</span>
<span class="co">#&gt; AAACACCAATAACTGC-1      151673      1  59  19 427.7678 183.0783       8    600</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1      151673      1  14  94 186.8137 417.2367       6    600</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1      151673      1  43   9 341.2691 152.7003       3    600</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1      151673      1  47  13 362.9163 164.9415       3    600</span>
<span class="co">#&gt;                    width sum_umi sum_gene subject position replicate</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1   600    8458     3586  Br8100        0         1</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1   600    1667     1150  Br8100        0         1</span>
<span class="co">#&gt; AAACACCAATAACTGC-1   600    3769     1960  Br8100        0         1</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1   600    5433     2424  Br8100        0         1</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1   600    4278     2264  Br8100        0         1</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1   600    4004     2178  Br8100        0         1</span>
<span class="co">#&gt;                    subject_position discard                       key</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1      Br8100_pos0   FALSE 151673_AAACAAGTATCTCCCA-1</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1      Br8100_pos0   FALSE 151673_AAACAATCTACTAGCA-1</span>
<span class="co">#&gt; AAACACCAATAACTGC-1      Br8100_pos0   FALSE 151673_AAACACCAATAACTGC-1</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1      Br8100_pos0   FALSE 151673_AAACAGAGCGACTCCT-1</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1      Br8100_pos0   FALSE 151673_AAACAGCTTTCAGAAG-1</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1      Br8100_pos0   FALSE 151673_AAACAGGGTCTATATT-1</span>
<span class="co">#&gt;                    cell_count SNN_k50_k4 SNN_k50_k5 SNN_k50_k6 SNN_k50_k7</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1          6          1          1          1          1</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1         16          1          1          1          1</span>
<span class="co">#&gt; AAACACCAATAACTGC-1          5          2          3          4          5</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1          2          1          1          1          1</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1          4          1          1          1          1</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1          6          2          4          5          6</span>
<span class="co">#&gt;                    SNN_k50_k8 SNN_k50_k9 SNN_k50_k10 SNN_k50_k11 SNN_k50_k12</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1          1          1           1           4           4</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1          1          1           1           4           4</span>
<span class="co">#&gt; AAACACCAATAACTGC-1          8          9          10          11          10</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1          1          1           1           2           2</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1          1          3           3           3           3</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1          5          6           7           8          12</span>
<span class="co">#&gt;                    SNN_k50_k13 SNN_k50_k14 SNN_k50_k15 SNN_k50_k16 SNN_k50_k17</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1           4           4           3           3           3</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1           4           4           3           3           3</span>
<span class="co">#&gt; AAACACCAATAACTGC-1           9          10          11          11          12</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1           2           2           1           1           1</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1           3           3           2           2           2</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1          12          13          14          15          16</span>
<span class="co">#&gt;                    SNN_k50_k18 SNN_k50_k19 SNN_k50_k20 SNN_k50_k21 SNN_k50_k22</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1           2           2           2           2           1</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1           2           2           2           2           1</span>
<span class="co">#&gt; AAACACCAATAACTGC-1          13          12          11          11          12</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1           4           4           4           4           3</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1           1           1           1           1           6</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1          17          17          17          16          17</span>
<span class="co">#&gt;                    SNN_k50_k23 SNN_k50_k24 SNN_k50_k25 SNN_k50_k26 SNN_k50_k27</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1           1           1           6           5           5</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1           1           1          10          10          10</span>
<span class="co">#&gt; AAACACCAATAACTGC-1          20          20          21          22          21</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1           3           3           2           2           2</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1           6           6           5           4           4</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1          16          16          17          18          17</span>
<span class="co">#&gt;                    SNN_k50_k28 GraphBased Maynard Martinowich Layer layer_guess</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1           4          7     2_3           2    NA      Layer3</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1          10          4     2_3           3    NA      Layer1</span>
<span class="co">#&gt; AAACACCAATAACTGC-1          22          8      WM          WM    NA          WM</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1           7          6       6           6    NA      Layer3</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1           3          3       5           5    NA      Layer5</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1          18          3       5           5    NA      Layer6</span>
<span class="co">#&gt;                    layer_guess_reordered layer_guess_reordered_short expr_chrM</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1                Layer3                          L3      1407</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1                Layer1                          L1       204</span>
<span class="co">#&gt; AAACACCAATAACTGC-1                    WM                          WM       430</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1                Layer3                          L3      1316</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1                Layer5                          L5       651</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1                Layer6                          L6       621</span>
<span class="co">#&gt;                    expr_chrM_ratio SpatialDE_PCA SpatialDE_pool_PCA HVG_PCA</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1       0.1663514             3                  3       2</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1       0.1223755             2                  5       3</span>
<span class="co">#&gt; AAACACCAATAACTGC-1       0.1140886             4                  4       7</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1       0.2422234             3                  3       2</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1       0.1521739             2                  1       1</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1       0.1550949             6                  7       8</span>
<span class="co">#&gt;                    pseudobulk_PCA markers_PCA SpatialDE_UMAP</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1              2           2              1</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1              7           8              2</span>
<span class="co">#&gt; AAACACCAATAACTGC-1              5           5              5</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1              3           6              1</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1              4           2              1</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1              8           4              1</span>
<span class="co">#&gt;                    SpatialDE_pool_UMAP HVG_UMAP pseudobulk_UMAP markers_UMAP</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1                   1        3               1            1</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1                   2        2               2            1</span>
<span class="co">#&gt; AAACACCAATAACTGC-1                   5        6               6            1</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1                   1        3               1            3</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1                   1        7               3            4</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1                   7        1               3            4</span>
<span class="co">#&gt;                    SpatialDE_PCA_spatial SpatialDE_pool_PCA_spatial</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1                     3                          1</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1                     7                          5</span>
<span class="co">#&gt; AAACACCAATAACTGC-1                     5                          4</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1                     3                          3</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1                     2                          1</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1                     6                          8</span>
<span class="co">#&gt;                    HVG_PCA_spatial pseudobulk_PCA_spatial markers_PCA_spatial</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1               1                      3                   1</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1               2                      2                   3</span>
<span class="co">#&gt; AAACACCAATAACTGC-1               4                      5                   3</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1               1                      2                   2</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1               2                      4                   1</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1               7                      7                   7</span>
<span class="co">#&gt;                    SpatialDE_UMAP_spatial SpatialDE_pool_UMAP_spatial</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1                      7                           1</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1                      2                           1</span>
<span class="co">#&gt; AAACACCAATAACTGC-1                      5                           7</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1                      3                           4</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1                      3                           3</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1                      3                           3</span>
<span class="co">#&gt;                    HVG_UMAP_spatial pseudobulk_UMAP_spatial</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1                1                       2</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1                4                       2</span>
<span class="co">#&gt; AAACACCAATAACTGC-1                5                       3</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1                2                       1</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1                8                       4</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1                3                       4</span>
<span class="co">#&gt;                    markers_UMAP_spatial sizeFactor x_gmm     r_gmm.V1</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1                    1  2.6257715     1 6.148291e-01</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1                    3  0.5269265     4 9.639893e-04</span>
<span class="co">#&gt; AAACACCAATAACTGC-1                    2  1.0484921     5 2.420481e-59</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1                    1  1.4784490     1 9.126456e-01</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1                    4  1.3386770     3 4.862401e-04</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1                    4  1.1790057     7 5.331015e-12</span>
<span class="co">#&gt;                        r_gmm.V2     r_gmm.V3     r_gmm.V4     r_gmm.V5</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1 2.720022e-24 3.373629e-03 4.191687e-05 1.159537e-49</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1 9.387657e-21 7.717454e-04 9.916840e-01 2.504924e-42</span>
<span class="co">#&gt; AAACACCAATAACTGC-1 3.239627e-12 2.125771e-51 4.454654e-56 1.000000e+00</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1 1.816712e-24 5.985415e-04 3.034282e-05 1.093182e-49</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1 1.298154e-13 8.302244e-01 1.998361e-10 2.101466e-36</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1 1.795491e-03 4.993974e-05 7.770341e-16 1.156953e-19</span>
<span class="co">#&gt;                        r_gmm.V6     r_gmm.V7 x_icm    r_icm.V1    r_icm.V2</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1 3.817553e-01 1.493637e-08     6 0.216099294 0.029245859</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1 6.580215e-03 5.273170e-08     4 0.002442427 0.002442427</span>
<span class="co">#&gt; AAACACCAATAACTGC-1 7.001587e-58 7.765638e-38     5 0.002442427 0.002442427</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1 8.672553e-02 1.234249e-08     6 0.002442427 0.002442427</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1 5.664366e-03 1.636250e-01     2 0.007702348 0.935915561</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1 1.319611e-11 9.981546e-01     2 0.002442427 0.985345437</span>
<span class="co">#&gt;                       r_icm.V3    r_icm.V4    r_icm.V5    r_icm.V6    r_icm.V7</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1 0.029245859 0.079498487 0.029245859 0.587418783 0.029245859</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1 0.002442427 0.985345437 0.002442427 0.002442427 0.002442427</span>
<span class="co">#&gt; AAACACCAATAACTGC-1 0.002442427 0.002442427 0.985345437 0.002442427 0.002442427</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1 0.002442427 0.002442427 0.002442427 0.985345437 0.002442427</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1 0.025572697 0.007702348 0.007702348 0.007702348 0.007702348</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1 0.002442427 0.002442427 0.002442427 0.002442427 0.002442427</span>
<span class="co">#&gt;                    x_gc     r_gc.V1     r_gc.V2     r_gc.V3     r_gc.V4</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1    6 0.216099294 0.029245859 0.029245859 0.079498487</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1    4 0.002442427 0.002442427 0.002442427 0.985345437</span>
<span class="co">#&gt; AAACACCAATAACTGC-1    5 0.002442427 0.002442427 0.002442427 0.002442427</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1    6 0.002442427 0.002442427 0.002442427 0.002442427</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1    7 0.002442427 0.002442427 0.002442427 0.002442427</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1    7 0.014928224 0.110305483 0.014928224 0.014928224</span>
<span class="co">#&gt;                        r_gc.V5     r_gc.V6     r_gc.V7</span>
<span class="co">#&gt; AAACAAGTATCTCCCA-1 0.029245859 0.587418783 0.029245859</span>
<span class="co">#&gt; AAACAATCTACTAGCA-1 0.002442427 0.002442427 0.002442427</span>
<span class="co">#&gt; AAACACCAATAACTGC-1 0.985345437 0.002442427 0.002442427</span>
<span class="co">#&gt; AAACAGAGCGACTCCT-1 0.002442427 0.985345437 0.002442427</span>
<span class="co">#&gt; AAACAGCTTTCAGAAG-1 0.002442427 0.002442427 0.985345437</span>
<span class="co">#&gt; AAACAGGGTCTATATT-1 0.014928224 0.014928224 0.815053399</span>

<span class="co">## Get the meta data of each spot for each data batch</span>
<span class="va">metadataList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">dlpfc2</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span>


<span class="co">## ensure the row.names of metadata in metaList are the same as that of colnames count matrix in countList</span>
<span class="kw">for</span><span class="op">(</span><span class="va">r</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">M</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">metadataList</span><span class="op">[[</span><span class="va">r</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">countList</span><span class="op">[[</span><span class="va">r</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="op">}</span>


<span class="co">## Create the Seurat list  object</span>

<span class="va">seuList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>
<span class="kw">for</span><span class="op">(</span><span class="va">r</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">M</span><span class="op">)</span><span class="op">{</span>
  <span class="va">seuList</span><span class="op">[[</span><span class="va">r</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">countList</span><span class="op">[[</span><span class="va">r</span><span class="op">]</span><span class="op">]</span>, meta.data<span class="op">=</span><span class="va">metadataList</span><span class="op">[[</span><span class="va">r</span><span class="op">]</span><span class="op">]</span>, project <span class="op">=</span> <span class="st">"ProFASTdlpfc2"</span><span class="op">)</span>
<span class="op">}</span>
<span class="co">#&gt; Warning: Non-unique features (rownames) present in the input matrix, making</span>
<span class="co">#&gt; unique</span>

<span class="co">#&gt; Warning: Non-unique features (rownames) present in the input matrix, making</span>
<span class="co">#&gt; unique</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">seuList</span><span class="op">)</span>
<span class="co">#&gt; [[1]]</span>
<span class="co">#&gt; An object of class Seurat </span>
<span class="co">#&gt; 15027 features across 3639 samples within 1 assay </span>
<span class="co">#&gt; Active assay: RNA (15027 features, 0 variable features)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]]</span>
<span class="co">#&gt; An object of class Seurat </span>
<span class="co">#&gt; 15809 features across 3673 samples within 1 assay </span>
<span class="co">#&gt; Active assay: RNA (15809 features, 0 variable features)</span>
<span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">seuList</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>
<span class="co">#&gt;  [1] "LINC01409"       "FAM41C"          "SAMD11"          "NOC2L"          </span>
<span class="co">#&gt;  [5] "KLHL17"          "ENSG00000272512" "HES4"            "ISG15"          </span>
<span class="co">#&gt;  [9] "AGRN"            "C1ORF159"</span></code></pre></div>
<div class="section level3">
<h3 id="prepare-the-precastobject-with-preprocessing-step-">Prepare the PRECASTObject with preprocessing step.<a class="anchor" aria-label="anchor" href="#prepare-the-precastobject-with-preprocessing-step-"></a>
</h3>
<p>Next, we use <code><a href="https://rdrr.io/pkg/PRECAST/man/CreateiDRSCObject.html" class="external-link">CreatePRECASTObject()</a></code> to create a PRECASTObject object based on the Seurat list object <code>seuList</code>. Here, we select the top 2000 highly variable genes for followed analysis. Users are able to see <a href="https://feiyoung.github.io/PRECAST/articles/PRECAST.BreastCancer.html" class="external-link uri">https://feiyoung.github.io/PRECAST/articles/PRECAST.BreastCancer.html</a> for what is done in this function.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Create PRECASTObject</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">2023</span><span class="op">)</span>
<span class="va">PRECASTObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PRECAST/man/CreateiDRSCObject.html" class="external-link">CreatePRECASTObject</a></span><span class="op">(</span><span class="va">seuList</span>, project <span class="op">=</span> <span class="st">"ProFASTdlpfc2"</span>, gene.number <span class="op">=</span> <span class="fl">2000</span>, selectGenesMethod <span class="op">=</span> <span class="st">"HVGs"</span>,
    premin.spots <span class="op">=</span> <span class="fl">20</span>, premin.features <span class="op">=</span> <span class="fl">20</span>, postmin.spots <span class="op">=</span> <span class="fl">1</span>, postmin.features <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>
<span class="co">#&gt; Filter spots and features from Raw count data...</span>
<span class="co">#&gt;  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Select common top variable genes  for multiple samples...</span>
<span class="co">#&gt; Filter spots and features from SVGs(HVGs) count data...</span>

<span class="co">## User can retain the raw seuList by the following commond.</span>
<span class="co">##  PRECASTObj &lt;-  CreatePRECASTObject(seuList, customGenelist=row.names(seuList[[1]]), rawData.preserve = TRUE)</span>
<span class="co">## check the number of genes/features after filtering step</span>
<span class="va">PRECASTObj</span><span class="op">@</span><span class="va">seulist</span>
<span class="co">#&gt; [[1]]</span>
<span class="co">#&gt; An object of class Seurat </span>
<span class="co">#&gt; 2000 features across 3639 samples within 1 assay </span>
<span class="co">#&gt; Active assay: RNA (2000 features, 1502 variable features)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]]</span>
<span class="co">#&gt; An object of class Seurat </span>
<span class="co">#&gt; 2000 features across 3673 samples within 1 assay </span>
<span class="co">#&gt; Active assay: RNA (2000 features, 1457 variable features)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="fit-precast-for-this-data">Fit PRECAST for this data<a class="anchor" aria-label="anchor" href="#fit-precast-for-this-data"></a>
</h2>
<div class="section level3">
<h3 id="add-the-model-setting">Add the model setting<a class="anchor" aria-label="anchor" href="#add-the-model-setting"></a>
</h3>
<p>Add adjacency matrix list and parameter setting of ProFAST. More model setting parameters can be found in <code><a href="../reference/model_set_ProFAST.html">model_set_ProFAST()</a></code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co">## seuList is null since the default value `rawData.preserve` is FALSE.</span>
<span class="va">PRECASTObj</span><span class="op">@</span><span class="va">seuList</span>
<span class="co">#&gt; NULL</span>

<span class="co">## Add adjacency matrix list for a PRECASTObj object to prepare for ProFAST model fitting.</span>
<span class="va">PRECASTObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PRECAST/man/AddAdjList.html" class="external-link">AddAdjList</a></span><span class="op">(</span><span class="va">PRECASTObj</span>, platform <span class="op">=</span> <span class="st">"Visium"</span><span class="op">)</span>
<span class="co">#&gt; Neighbors were identified for 3638 out of 3639 spots.</span>
<span class="co">#&gt; Neighbors were identified for 3670 out of 3673 spots.</span>

<span class="co">## Add a model setting in advance for a PRECASTObj object: verbose =TRUE helps outputing the information in the algorithm; </span>
<span class="va">PRECASTObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AddParSettingProFAST.html">AddParSettingProFAST</a></span><span class="op">(</span><span class="va">PRECASTObj</span>, verbose<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="co">## Check the parameters</span>
<span class="va">PRECASTObj</span><span class="op">@</span><span class="va">parameterList</span>
<span class="co">#&gt; $maxIter</span>
<span class="co">#&gt; [1] 30</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $seed</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $epsLogLik</span>
<span class="co">#&gt; [1] 1e-05</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $verbose</span>
<span class="co">#&gt; [1] TRUE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $error_heter</span>
<span class="co">#&gt; [1] TRUE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Psi_diag</span>
<span class="co">#&gt; [1] FALSE</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="fit-profast">Fit ProFAST<a class="anchor" aria-label="anchor" href="#fit-profast"></a>
</h3>
<p>For function <code>ProFAST</code>, users can specify the number of factors <code>q</code> and the fitted model <code>fit.model</code>. The <code>q</code> sets the number of spatial factors to be extracted, and a lareger one means more information to be extracted but higher computaional cost. The <code>fit.model</code> specifies the version of ProFAST to be fitted. The Gaussian version (<code>gaussian</code>) models the log-count matrices while the Poisson verion (<code>poisson</code>) models the count matrices; default as <code>poisson</code>. (Note: The computational time required to run the analysis on personal PCs is approximately 4.5 minutes.)</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">### set q= 15 here</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">2023</span><span class="op">)</span>
<span class="va">PRECASTObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ProFAST.html">ProFAST</a></span><span class="op">(</span><span class="va">PRECASTObj</span>, q<span class="op">=</span><span class="fl">15</span><span class="op">)</span>
<span class="co">#&gt; ******Run the Poisson version of ProFAST...</span>
<span class="co">#&gt; Loading required package: irlba</span>
<span class="co">#&gt; Loading required package: Matrix</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'Matrix'</span>
<span class="co">#&gt; The following object is masked from 'package:S4Vectors':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand</span>
<span class="co">#&gt; Finish variable initialization </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 2, ELBO= -411372079.669657, dELBO=0.808440 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 3, ELBO= -17213794.763737, dELBO=0.958155 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 4, ELBO= -16605861.092513, dELBO=0.035317 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 5, ELBO= -16176797.251864, dELBO=0.025838 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 6, ELBO= -15830109.692300, dELBO=0.021431 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 7, ELBO= -15537935.545724, dELBO=0.018457 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 8, ELBO= -15285188.986451, dELBO=0.016266 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 9, ELBO= -15062500.384941, dELBO=0.014569 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 10, ELBO= -14863550.995611, dELBO=0.013208 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 11, ELBO= -14683847.548923, dELBO=0.012090 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 12, ELBO= -14520071.178546, dELBO=0.011154 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 13, ELBO= -14369696.891473, dELBO=0.010356 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 14, ELBO= -14230757.378262, dELBO=0.009669 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 15, ELBO= -14101689.173611, dELBO=0.009070 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 16, ELBO= -13981228.819648, dELBO=0.008542 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 17, ELBO= -13868340.529789, dELBO=0.008074 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 18, ELBO= -13762164.519302, dELBO=0.007656 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 19, ELBO= -13661979.241551, dELBO=0.007280 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 20, ELBO= -13567173.264411, dELBO=0.006939 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 21, ELBO= -13477223.955954, dELBO=0.006630 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 22, ELBO= -13391681.094478, dELBO=0.006347 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 23, ELBO= -13310154.089707, dELBO=0.006088 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 24, ELBO= -13232301.904884, dELBO=0.005849 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 25, ELBO= -13157825.018383, dELBO=0.005628 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 26, ELBO= -13086458.953119, dELBO=0.005424 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 27, ELBO= -13017969.017104, dELBO=0.005234 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 28, ELBO= -12952145.996615, dELBO=0.005056 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 29, ELBO= -12888802.598514, dELBO=0.004891 </span>
<span class="co">#&gt; Satrt ICM step! </span>
<span class="co">#&gt; iter = 30, ELBO= -12827770.493728, dELBO=0.004735</span>
<span class="co">#&gt; 2023-06-04 18:36:23 : ***** Finish ProFAST, 3.565 mins elapsed.</span>
<span class="co">### Check the results</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">PRECASTObj</span><span class="op">@</span><span class="va">resList</span><span class="op">)</span>
<span class="co">#&gt; List of 1</span>
<span class="co">#&gt;  $ ProFAST:List of 7</span>
<span class="co">#&gt;   ..$ hV      :List of 2</span>
<span class="co">#&gt;   .. ..$ : num [1:3639, 1:15] 3.95 -29.47 -30.76 -8.14 -12.27 ...</span>
<span class="co">#&gt;   .. ..$ : num [1:3673, 1:15] 4.36 -17.23 -28.38 -3.41 -20.61 ...</span>
<span class="co">#&gt;   .. ..- attr(*, "dim")= int [1:2] 2 1</span>
<span class="co">#&gt;   ..$ nu      : num [1:2, 1:2000] -0.208 0.115 3.449 3.68 2.009 ...</span>
<span class="co">#&gt;   ..$ Psi     : num [1:15, 1:15, 1:2] 51.73 17.22 -7.83 1.49 -6.12 ...</span>
<span class="co">#&gt;   ..$ W       : num [1:2000, 1:15] 0.01463 -0.01035 0.01177 0.00159 0.00805 ...</span>
<span class="co">#&gt;   ..$ Lam     : num [1:2, 1:2000] 1.132 0.8974 0.1194 0.0939 0.924 ...</span>
<span class="co">#&gt;   ..$ ELBO    : num -12827770</span>
<span class="co">#&gt;   ..$ ELBO_seq: num [1:30] -2.15e+09 -4.11e+08 -1.72e+07 -1.66e+07 -1.62e+07 ...</span></code></pre></div>
<details><p><summary><strong>Run gaussian version</strong></summary> Users can also use the gaussian version by the following command:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">2023</span><span class="op">)</span>
<span class="va">PRECASTObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ProFAST.html">ProFAST</a></span><span class="op">(</span><span class="va">PRECASTObj</span>, q<span class="op">=</span><span class="fl">15</span>, fit.model<span class="op">=</span><span class="st">'gaussian'</span><span class="op">)</span>
<span class="co">### Check the results</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">PRECASTObj</span><span class="op">@</span><span class="va">resList</span><span class="op">)</span></code></pre></div>
</details>
</div>
<div class="section level3">
<h3 id="evaluate-the-adjusted-mcfaddens-pseudo-r-square">Evaluate the adjusted McFadden’s pseudo R-square<a class="anchor" aria-label="anchor" href="#evaluate-the-adjusted-mcfaddens-pseudo-r-square"></a>
</h3>
<p>Next, we investigate the performance of dimension reduction by calculating the adjusted McFadden’s pseudo R-square for each data batch. The manual annotations are regarded as the groud truth in the <code>meta.data</code> of each component of <code>PRECASTObj@seulist</code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Obtain the true labels</span>
<span class="va">yList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">PRECASTObj</span><span class="op">@</span><span class="va">seulist</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">$</span><span class="va">layer_guess_reordered</span><span class="op">)</span>
<span class="co">### Evaluate the MacR2</span>
<span class="va">MacVec</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">PRECASTObj</span><span class="op">@</span><span class="va">seulist</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">r</span><span class="op">)</span> <span class="fu"><a href="../reference/get_r2_mcfadden.html">get_r2_mcfadden</a></span><span class="op">(</span><span class="va">PRECASTObj</span><span class="op">@</span><span class="va">resList</span><span class="op">$</span><span class="va">ProFAST</span><span class="op">$</span><span class="va">hV</span><span class="op">[[</span><span class="va">r</span><span class="op">]</span><span class="op">]</span>, <span class="va">yList</span><span class="op">[[</span><span class="va">r</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; # weights:  119 (96 variable)</span>
<span class="co">#&gt; initial  value 7026.681548 </span>
<span class="co">#&gt; iter  10 value 2129.810642</span>
<span class="co">#&gt; iter  20 value 1774.882302</span>
<span class="co">#&gt; iter  30 value 1702.705338</span>
<span class="co">#&gt; iter  40 value 1659.201724</span>
<span class="co">#&gt; iter  50 value 1508.463538</span>
<span class="co">#&gt; iter  60 value 1218.670146</span>
<span class="co">#&gt; iter  70 value 998.999606</span>
<span class="co">#&gt; iter  80 value 879.397400</span>
<span class="co">#&gt; iter  90 value 750.966769</span>
<span class="co">#&gt; iter 100 value 663.665727</span>
<span class="co">#&gt; final  value 663.665727 </span>
<span class="co">#&gt; stopped after 100 iterations</span>
<span class="co">#&gt; # weights:  119 (96 variable)</span>
<span class="co">#&gt; initial  value 7073.383392 </span>
<span class="co">#&gt; iter  10 value 2159.681358</span>
<span class="co">#&gt; iter  20 value 1218.651591</span>
<span class="co">#&gt; iter  30 value 1112.886781</span>
<span class="co">#&gt; iter  40 value 1044.573888</span>
<span class="co">#&gt; iter  50 value 950.715777</span>
<span class="co">#&gt; iter  60 value 868.038840</span>
<span class="co">#&gt; iter  70 value 749.329895</span>
<span class="co">#&gt; iter  80 value 711.661631</span>
<span class="co">#&gt; iter  90 value 654.184634</span>
<span class="co">#&gt; iter 100 value 560.384499</span>
<span class="co">#&gt; final  value 560.384499 </span>
<span class="co">#&gt; stopped after 100 iterations</span>
<span class="co">### output them</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">MacVec</span><span class="op">)</span>
<span class="co">#&gt; adjusted McFadden's R2 adjusted McFadden's R2 </span>
<span class="co">#&gt;              0.8984519              0.9162326</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="embedding-alignment-and-spatial-clustering-using-isc-meb">Embedding alignment and spatial clustering using iSC-MEB<a class="anchor" aria-label="anchor" href="#embedding-alignment-and-spatial-clustering-using-isc-meb"></a>
</h2>
<p>Based on the embeddings from ProFAST, we use <code>iSC-MEB</code> to jointly align the embeddings and perform clustering. In this downstream analysis, other methods for embedding alignment and clustering can be also used. In the vignette of the simulated ST data, we show another method (<code>Harmony</code>+<code>Louvain</code>) to perform embedding alignment and clustering.</p>
<p>First, we use the function <code><a href="../reference/RunHarmonyLouvain.html">RunHarmonyLouvain()</a></code> in <code>ProFAST</code> R package to select the number of clusters</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">PRECASTObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/RunHarmonyLouvain.html">RunHarmonyLouvain</a></span><span class="op">(</span><span class="va">PRECASTObj</span>, resolution <span class="op">=</span> <span class="fl">0.3</span><span class="op">)</span>
<span class="co">#&gt; ******Use Harmony to remove batch in the embeddings from ProFAST...</span>
<span class="co">#&gt; Harmony 1/10</span>
<span class="co">#&gt; Harmony 2/10</span>
<span class="co">#&gt; Harmony 3/10</span>
<span class="co">#&gt; Harmony converged after 3 iterations</span>
<span class="co">#&gt; 2023-06-04 18:36:30 : ***** Finish Harmony correction, 0.09 mins elapsed.</span>
<span class="co">#&gt; ******Use Louvain to cluster and determine the number of clusters...</span>
<span class="co">#&gt; Warning: No assay specified, setting assay as RNA by default.</span>
<span class="co">#&gt; Warning: Keys should be one or more alphanumeric characters followed by an</span>
<span class="co">#&gt; underscore, setting key from PC to PC_</span>
<span class="co">#&gt; Warning: All keys should be one or more alphanumeric characters followed by an</span>
<span class="co">#&gt; underscore '_', setting key to PC_</span>
<span class="co">#&gt; Computing nearest neighbor graph</span>
<span class="co">#&gt; Computing SNN</span>
<span class="co">#&gt; Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Number of nodes: 7312</span>
<span class="co">#&gt; Number of edges: 208186</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Running Louvain algorithm...</span>
<span class="co">#&gt; Maximum modularity in 10 random starts: 0.9132</span>
<span class="co">#&gt; Number of communities: 9</span>
<span class="co">#&gt; Elapsed time: 0 seconds</span>
<span class="co">#&gt; 2023-06-04 18:36:35 : ***** Finsh Louvain clustering and find the optimal number of clusters, 0.076 mins elapsed.</span>
<span class="va">ARI_vec_louvain</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="va">M</span>, <span class="kw">function</span><span class="op">(</span><span class="va">r</span><span class="op">)</span> <span class="fu">mclust</span><span class="fu">::</span><span class="fu"><a href="https://mclust-org.github.io/mclust/reference/adjustedRandIndex.html" class="external-link">adjustedRandIndex</a></span><span class="op">(</span><span class="va">PRECASTObj</span><span class="op">@</span><span class="va">resList</span><span class="op">$</span><span class="va">Louvain</span><span class="op">$</span><span class="va">cluster</span><span class="op">[[</span><span class="va">r</span><span class="op">]</span><span class="op">]</span>, <span class="va">yList</span><span class="op">[[</span><span class="va">r</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">ARI_vec_louvain</span><span class="op">)</span>
<span class="co">#&gt; [1] 0.4040998 0.4416045</span></code></pre></div>
<p>Then, we use the function <code><a href="../reference/RuniSCMEB.html">RuniSCMEB()</a></code> <code>ProFAST</code> R package to jointly perform embedding alignment and spatial clustering. The results are saved in the slot <code>PRECASTObj@resList$iSCMEB</code>.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" title="1">PRECASTObj &lt;-<span class="st"> </span><span class="kw">RuniSCMEB</span>(PRECASTObj, <span class="dt">seed=</span><span class="dv">1</span>)</a>
<a class="sourceLine" id="cb14-2" title="2"><span class="co">#&gt; ******Perform embedding alignment and spatial clustering using iSC-MEB based on  the embeddings obtained by ProFAST...</span></a>
<a class="sourceLine" id="cb14-3" title="3"><span class="co">#&gt; Evaluate initial values...</span></a>
<a class="sourceLine" id="cb14-4" title="4"><span class="co">#&gt; Loading required package: mclust</span></a>
<a class="sourceLine" id="cb14-5" title="5"><span class="co">#&gt; Warning: package 'mclust' was built under R version 4.1.3</span></a>
<a class="sourceLine" id="cb14-6" title="6"><span class="co">#&gt; Package 'mclust' version 5.4.10</span></a>
<a class="sourceLine" id="cb14-7" title="7"><span class="co">#&gt; Type 'citation("mclust")' for citing this R package in publications.</span></a>
<a class="sourceLine" id="cb14-8" title="8"><span class="co">#&gt; fitting ...</span></a>
<a class="sourceLine" id="cb14-9" title="9"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb14-10" title="10">  <span class="op">|</span><span class="st">                                                                            </span></a>
<a class="sourceLine" id="cb14-11" title="11"><span class="st">  </span><span class="er">|</span><span class="st">                                                                      </span><span class="er">|</span><span class="st">   </span><span class="dv">0</span>%</a>
<a class="sourceLine" id="cb14-12" title="12">  <span class="op">|</span><span class="st">                                                                            </span></a>
<a class="sourceLine" id="cb14-13" title="13"><span class="st">  </span><span class="er">|===================================</span><span class="st">                                   </span><span class="er">|</span><span class="st">  </span><span class="dv">50</span>%</a>
<a class="sourceLine" id="cb14-14" title="14">  <span class="op">|</span><span class="st">                                                                            </span></a>
<a class="sourceLine" id="cb14-15" title="15"><span class="st">  </span><span class="er">|======================================================================|</span><span class="st"> </span><span class="dv">100</span>%</a>
<a class="sourceLine" id="cb14-16" title="16"><span class="co">#&gt; Fit SC-MEB2...</span></a>
<a class="sourceLine" id="cb14-17" title="17"><span class="co">#&gt; Finish variable initialization </span></a>
<a class="sourceLine" id="cb14-18" title="18"><span class="co">#&gt; K = 9, iter = 2, loglik= -150896.002573, dloglik=0.999930 </span></a>
<a class="sourceLine" id="cb14-19" title="19"><span class="co">#&gt; K = 9, iter = 3, loglik= -124476.112108, dloglik=0.175087 </span></a>
<a class="sourceLine" id="cb14-20" title="20"><span class="co">#&gt; K = 9, iter = 4, loglik= -101972.965259, dloglik=0.180783 </span></a>
<a class="sourceLine" id="cb14-21" title="21"><span class="co">#&gt; K = 9, iter = 5, loglik= -83202.495447, dloglik=0.184073 </span></a>
<a class="sourceLine" id="cb14-22" title="22"><span class="co">#&gt; K = 9, iter = 6, loglik= -67251.132173, dloglik=0.191717 </span></a>
<a class="sourceLine" id="cb14-23" title="23"><span class="co">#&gt; K = 9, iter = 7, loglik= -54755.326181, dloglik=0.185808 </span></a>
<a class="sourceLine" id="cb14-24" title="24"><span class="co">#&gt; K = 9, iter = 8, loglik= -41834.846163, dloglik=0.235968 </span></a>
<a class="sourceLine" id="cb14-25" title="25"><span class="co">#&gt; K = 9, iter = 9, loglik= -33525.470987, dloglik=0.198623 </span></a>
<a class="sourceLine" id="cb14-26" title="26"><span class="co">#&gt; K = 9, iter = 10, loglik= -27065.801650, dloglik=0.192679 </span></a>
<a class="sourceLine" id="cb14-27" title="27"><span class="co">#&gt; K = 9, iter = 11, loglik= -21468.706053, dloglik=0.206796 </span></a>
<a class="sourceLine" id="cb14-28" title="28"><span class="co">#&gt; K = 9, iter = 12, loglik= -17626.350996, dloglik=0.178975 </span></a>
<a class="sourceLine" id="cb14-29" title="29"><span class="co">#&gt; K = 9, iter = 13, loglik= -14572.939088, dloglik=0.173230 </span></a>
<a class="sourceLine" id="cb14-30" title="30"><span class="co">#&gt; K = 9, iter = 14, loglik= -12418.910316, dloglik=0.147810 </span></a>
<a class="sourceLine" id="cb14-31" title="31"><span class="co">#&gt; K = 9, iter = 15, loglik= -10791.681004, dloglik=0.131028 </span></a>
<a class="sourceLine" id="cb14-32" title="32"><span class="co">#&gt; K = 9, iter = 16, loglik= -9614.190635, dloglik=0.109111 </span></a>
<a class="sourceLine" id="cb14-33" title="33"><span class="co">#&gt; K = 9, iter = 17, loglik= -8760.079145, dloglik=0.088839 </span></a>
<a class="sourceLine" id="cb14-34" title="34"><span class="co">#&gt; K = 9, iter = 18, loglik= -7857.011708, dloglik=0.103089 </span></a>
<a class="sourceLine" id="cb14-35" title="35"><span class="co">#&gt; K = 9, iter = 19, loglik= -7233.435673, dloglik=0.079366 </span></a>
<a class="sourceLine" id="cb14-36" title="36"><span class="co">#&gt; K = 9, iter = 20, loglik= -6680.512477, dloglik=0.076440 </span></a>
<a class="sourceLine" id="cb14-37" title="37"><span class="co">#&gt; K = 9, iter = 21, loglik= -6279.045887, dloglik=0.060095 </span></a>
<a class="sourceLine" id="cb14-38" title="38"><span class="co">#&gt; K = 9, iter = 22, loglik= -5920.680543, dloglik=0.057073 </span></a>
<a class="sourceLine" id="cb14-39" title="39"><span class="co">#&gt; K = 9, iter = 23, loglik= -5740.179221, dloglik=0.030487 </span></a>
<a class="sourceLine" id="cb14-40" title="40"><span class="co">#&gt; K = 9, iter = 24, loglik= -5430.563137, dloglik=0.053938 </span></a>
<a class="sourceLine" id="cb14-41" title="41"><span class="co">#&gt; K = 9, iter = 25, loglik= -5438.098887, dloglik=0.001388</span></a>
<a class="sourceLine" id="cb14-42" title="42"><span class="co">#&gt; 2023-06-04 18:37:18 : ***** Finish iSC-MEB fitting, 0.722 mins elapsed.</span></a>
<a class="sourceLine" id="cb14-43" title="43"><span class="kw">str</span>(PRECASTObj<span class="op">@</span>resList<span class="op">$</span>iSCMEB)</a>
<a class="sourceLine" id="cb14-44" title="44"><span class="co">#&gt; List of 4</span></a>
<a class="sourceLine" id="cb14-45" title="45"><span class="co">#&gt;  $ cluster     :List of 2</span></a>
<a class="sourceLine" id="cb14-46" title="46"><span class="co">#&gt;   ..$ : num [1:3639] 2 9 7 2 6 6 9 2 5 1 ...</span></a>
<a class="sourceLine" id="cb14-47" title="47"><span class="co">#&gt;   ..$ : num [1:3673] 2 9 7 5 6 6 6 2 5 1 ...</span></a>
<a class="sourceLine" id="cb14-48" title="48"><span class="co">#&gt;  $ alignedEmbed:List of 2</span></a>
<a class="sourceLine" id="cb14-49" title="49"><span class="co">#&gt;   ..$ : num [1:3639, 1:15] -7.69 -23.49 -22.8 -7.2 -14.4 ...</span></a>
<a class="sourceLine" id="cb14-50" title="50"><span class="co">#&gt;   ..$ : num [1:3673, 1:15] -7.26 -21.97 -22.64 -5.7 -14.57 ...</span></a>
<a class="sourceLine" id="cb14-51" title="51"><span class="co">#&gt;   ..- attr(*, "dim")= int [1:2] 2 1</span></a>
<a class="sourceLine" id="cb14-52" title="52"><span class="co">#&gt;  $ batchEmbed  :List of 2</span></a>
<a class="sourceLine" id="cb14-53" title="53"><span class="co">#&gt;   ..$ : num [1:3639, 1:15] 3.95 -29.47 -30.76 -8.14 -12.27 ...</span></a>
<a class="sourceLine" id="cb14-54" title="54"><span class="co">#&gt;   ..$ : num [1:3673, 1:15] 4.36 -17.23 -28.38 -3.41 -20.61 ...</span></a>
<a class="sourceLine" id="cb14-55" title="55"><span class="co">#&gt;   ..- attr(*, "dim")= int [1:2] 2 1</span></a>
<a class="sourceLine" id="cb14-56" title="56"><span class="co">#&gt;  $ RList       :List of 2</span></a>
<a class="sourceLine" id="cb14-57" title="57"><span class="co">#&gt;   ..$ : num [1:3639, 1:9] 1.62e-68 3.87e-131 2.44e-71 3.32e-86 1.00e-29 ...</span></a>
<a class="sourceLine" id="cb14-58" title="58"><span class="co">#&gt;   ..$ : num [1:3673, 1:9] 1.40e-85 4.91e-139 2.11e-59 4.10e-142 8.31e-26 ...</span></a>
<a class="sourceLine" id="cb14-59" title="59"><span class="co">#&gt;   ..- attr(*, "dim")= int [1:2] 2 1</span></a>
<a class="sourceLine" id="cb14-60" title="60"><span class="kw">sapply</span>(<span class="dv">1</span><span class="op">:</span>M, <span class="cf">function</span>(r) mclust<span class="op">::</span><span class="kw">adjustedRandIndex</span>(PRECASTObj<span class="op">@</span>resList<span class="op">$</span>iSCMEB<span class="op">$</span>cluster[[r]], yList[[r]]))</a>
<a class="sourceLine" id="cb14-61" title="61"><span class="co">#&gt; [1] 0.4695632 0.4739501</span></a></code></pre></div>
</div>
<div class="section level2">
<h2 id="remove-unwanted-variations-in-the-log-normalized-expression-matrices">Remove unwanted variations in the log-normalized expression matrices<a class="anchor" aria-label="anchor" href="#remove-unwanted-variations-in-the-log-normalized-expression-matrices"></a>
</h2>
<p>In the following, we remove the unwanted variations in the log-normalized expression matrices to obtain a combined log-normalized expression matrix in a Seurat object. For this real data, we use housekeeping genes as negative control to remove the unwanted variations. Specifically, we leverage the batch effect embeddings estimated through housekeeping gene expression matrix to capture unwanted variations. Additionally, we utilize the cluster labels obtained via iSC-MEB to retain the desired biological effects. The posterior probability of cluster labels (RList) are in the slot <code>PRECASTObj@resList$iSCMEB</code>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">PRECASTObj</span><span class="op">@</span><span class="va">resList</span><span class="op">$</span><span class="va">iSCMEB</span><span class="op">)</span>
<span class="co">#&gt; List of 4</span>
<span class="co">#&gt;  $ cluster     :List of 2</span>
<span class="co">#&gt;   ..$ : num [1:3639] 2 9 7 2 6 6 9 2 5 1 ...</span>
<span class="co">#&gt;   ..$ : num [1:3673] 2 9 7 5 6 6 6 2 5 1 ...</span>
<span class="co">#&gt;  $ alignedEmbed:List of 2</span>
<span class="co">#&gt;   ..$ : num [1:3639, 1:15] -7.69 -23.49 -22.8 -7.2 -14.4 ...</span>
<span class="co">#&gt;   ..$ : num [1:3673, 1:15] -7.26 -21.97 -22.64 -5.7 -14.57 ...</span>
<span class="co">#&gt;   ..- attr(*, "dim")= int [1:2] 2 1</span>
<span class="co">#&gt;  $ batchEmbed  :List of 2</span>
<span class="co">#&gt;   ..$ : num [1:3639, 1:15] 3.95 -29.47 -30.76 -8.14 -12.27 ...</span>
<span class="co">#&gt;   ..$ : num [1:3673, 1:15] 4.36 -17.23 -28.38 -3.41 -20.61 ...</span>
<span class="co">#&gt;   ..- attr(*, "dim")= int [1:2] 2 1</span>
<span class="co">#&gt;  $ RList       :List of 2</span>
<span class="co">#&gt;   ..$ : num [1:3639, 1:9] 1.62e-68 3.87e-131 2.44e-71 3.32e-86 1.00e-29 ...</span>
<span class="co">#&gt;   ..$ : num [1:3673, 1:9] 1.40e-85 4.91e-139 2.11e-59 4.10e-142 8.31e-26 ...</span>
<span class="co">#&gt;   ..- attr(*, "dim")= int [1:2] 2 1</span></code></pre></div>
<p>To capture the unwanted variation, we first select the select the housekeeping genes using the function <code><a href="../reference/SelectHKgenes.html">SelectHKgenes()</a></code>.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## select the HK genes</span>
<span class="va">HKgenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SelectHKgenes.html">SelectHKgenes</a></span><span class="op">(</span><span class="va">seuList</span>, species<span class="op">=</span> <span class="st">"Human"</span>, HK.number<span class="op">=</span><span class="fl">200</span><span class="op">)</span>
<span class="co">#&gt; Find the spatially variables genes by SPARK-X...</span>
<span class="co">#&gt; ## ===== SPARK-X INPUT INFORMATION ====</span>
<span class="co">#&gt; ## number of total samples: 3639</span>
<span class="co">#&gt; ## number of total genes: 2107</span>
<span class="co">#&gt; ## Running with single core, may take some time</span>
<span class="co">#&gt; ## Testing With Projection Kernel</span>
<span class="co">#&gt; ## Testing With Gaussian Kernel 1</span>
<span class="co">#&gt; ## Testing With Gaussian Kernel 2</span>
<span class="co">#&gt; ## Testing With Gaussian Kernel 3</span>
<span class="co">#&gt; ## Testing With Gaussian Kernel 4</span>
<span class="co">#&gt; ## Testing With Gaussian Kernel 5</span>
<span class="co">#&gt; ## Testing With Cosine Kernel 1</span>
<span class="co">#&gt; ## Testing With Cosine Kernel 2</span>
<span class="co">#&gt; ## Testing With Cosine Kernel 3</span>
<span class="co">#&gt; ## Testing With Cosine Kernel 4</span>
<span class="co">#&gt; ## Testing With Cosine Kernel 5</span>
<span class="co">#&gt; Find the spatially variables genes by SPARK-X...</span>
<span class="co">#&gt; ## ===== SPARK-X INPUT INFORMATION ====</span>
<span class="co">#&gt; ## number of total samples: 3673</span>
<span class="co">#&gt; ## number of total genes: 2107</span>
<span class="co">#&gt; ## Running with single core, may take some time</span>
<span class="co">#&gt; ## Testing With Projection Kernel</span>
<span class="co">#&gt; ## Testing With Gaussian Kernel 1</span>
<span class="co">#&gt; ## Testing With Gaussian Kernel 2</span>
<span class="co">#&gt; ## Testing With Gaussian Kernel 3</span>
<span class="co">#&gt; ## Testing With Gaussian Kernel 4</span>
<span class="co">#&gt; ## Testing With Gaussian Kernel 5</span>
<span class="co">#&gt; ## Testing With Cosine Kernel 1</span>
<span class="co">#&gt; ## Testing With Cosine Kernel 2</span>
<span class="co">#&gt; ## Testing With Cosine Kernel 3</span>
<span class="co">#&gt; ## Testing With Cosine Kernel 4</span>
<span class="co">#&gt; ## Testing With Cosine Kernel 5</span>
<span class="va">seulist_HK</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">seuList</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">[</span><span class="va">HKgenes</span>, <span class="op">]</span><span class="op">)</span>
<span class="va">seulist_HK</span>
<span class="co">#&gt; [[1]]</span>
<span class="co">#&gt; An object of class Seurat </span>
<span class="co">#&gt; 200 features across 3639 samples within 1 assay </span>
<span class="co">#&gt; Active assay: RNA (200 features, 0 variable features)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]]</span>
<span class="co">#&gt; An object of class Seurat </span>
<span class="co">#&gt; 200 features across 3673 samples within 1 assay </span>
<span class="co">#&gt; Active assay: RNA (200 features, 0 variable features)</span></code></pre></div>
<p>Then, we integrate the two sections by removing the unwanted variations by using <code>seulist_HK=seulist_HK</code> and <code>Method = "iSC-MEB"</code> in the function <code><a href="../reference/IntegrateSRTData.html">IntegrateSRTData()</a></code>. After obtaining <code>seuInt</code>, we will see there are three embeddings: <code>profast</code>, <code>iscmeb</code> and <code>position</code>, in the slot <code>seuInt@reductions</code>. <code>profast</code> are the embeddings obtained by ProFAST model fitting and are uncorrected embeddings that may includes the unwanted effects (i.e., batch effects); <code>iscmeb</code>are the aligned embeddings obtained by iSC-MEB fitting; and <code>position</code> are the spatial coordinates.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seuInt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/IntegrateSRTData.html">IntegrateSRTData</a></span><span class="op">(</span><span class="va">PRECASTObj</span>, seulist_HK<span class="op">=</span><span class="va">seulist_HK</span>, Method <span class="op">=</span> <span class="st">"iSC-MEB"</span>,
                           seuList_raw<span class="op">=</span><span class="cn">NULL</span>, covariates_use<span class="op">=</span><span class="cn">NULL</span>, verbose<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; ******Perform PCA on housekeeping gene expression matrix...</span>
<span class="co">#&gt; Use housekeeping genes and the results in iSC-MEB to remove unwanted variations...</span>
<span class="co">#&gt; 2023-06-04 18:37:32 : ***** Finish PCA, 0.01 mins elapsed.</span>
<span class="co">#&gt; ******Remove the unwanted variations in gene expressions using spatial linear regression...</span>
<span class="co">#&gt; Finish variable initialization </span>
<span class="co">#&gt; iter = 2, ELBO= -15008760.721226, dELBO=0.993011 </span>
<span class="co">#&gt; iter = 3, ELBO= -12380105.756939, dELBO=0.175141 </span>
<span class="co">#&gt; iter = 4, ELBO= -10455078.042263, dELBO=0.155494 </span>
<span class="co">#&gt; iter = 5, ELBO= -9101289.791041, dELBO=0.129486 </span>
<span class="co">#&gt; iter = 6, ELBO= -8235213.614892, dELBO=0.095160 </span>
<span class="co">#&gt; iter = 7, ELBO= -7744831.177581, dELBO=0.059547 </span>
<span class="co">#&gt; iter = 8, ELBO= -7507465.403588, dELBO=0.030648 </span>
<span class="co">#&gt; iter = 9, ELBO= -7410473.167090, dELBO=0.012919 </span>
<span class="co">#&gt; iter = 10, ELBO= -7376536.502825, dELBO=0.004580 </span>
<span class="co">#&gt; iter = 11, ELBO= -7365797.029959, dELBO=0.001456 </span>
<span class="co">#&gt; iter = 12, ELBO= -7362426.830820, dELBO=0.000458 </span>
<span class="co">#&gt; iter = 13, ELBO= -7361180.424023, dELBO=0.000169 </span>
<span class="co">#&gt; iter = 14, ELBO= -7360533.357938, dELBO=0.000088</span>
<span class="co">#&gt; 2023-06-04 18:38:36 : ***** Finish unwanted variation removal, 1.053 mins elapsed.</span>
<span class="co">#&gt; ******Sort the results into a Seurat object...</span>
<span class="co">#&gt; 2023-06-04 18:38:38 : ***** Finish results arrangement, 0.032 mins elapsed.</span>
<span class="va">seuInt</span>
<span class="co">#&gt; An object of class Seurat </span>
<span class="co">#&gt; 2000 features across 7312 samples within 1 assay </span>
<span class="co">#&gt; Active assay: ProFAST (2000 features, 0 variable features)</span>
<span class="co">#&gt;  3 dimensional reductions calculated: profast, iscmeb, position</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<p>First, user can choose a beautiful color schema using <code><a href="https://rdrr.io/pkg/PRECAST/man/chooseColors.html" class="external-link">chooseColors()</a></code> in the R package <code>PRECAST</code>.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cols_cluster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PRECAST/man/chooseColors.html" class="external-link">chooseColors</a></span><span class="op">(</span>palettes_name <span class="op">=</span> <span class="st">"Nature 10"</span>, n_colors <span class="op">=</span> <span class="fl">9</span>, plot_colors <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="ProFASTdlpfc2_files/figure-html/unnamed-chunk-19-1.png" width="700"> Then, we plot the spatial scatter plot for clusters using the function <code><a href="https://rdrr.io/pkg/PRECAST/man/SpaPlot.html" class="external-link">SpaPlot()</a></code> in the R package <code>PRECAST</code>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p12</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PRECAST/man/SpaPlot.html" class="external-link">SpaPlot</a></span><span class="op">(</span><span class="va">seuInt</span>, item <span class="op">=</span> <span class="st">"cluster"</span>, batch <span class="op">=</span> <span class="cn">NULL</span>, point_size <span class="op">=</span> <span class="fl">1</span>, cols <span class="op">=</span> <span class="va">cols_cluster</span>, combine <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.1.3</span>
<span class="va">p12</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">p12</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_reverse</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/PRECAST/man/drawFigs.html" class="external-link">drawFigs</a></span><span class="op">(</span><span class="va">p12</span>, layout.dim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span>, common.legend <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<p><img src="ProFASTdlpfc2_files/figure-html/unnamed-chunk-20-1.png" width="624"></p>
<p>We use the function <code><a href="https://rdrr.io/pkg/PRECAST/man/AddUMAP.html" class="external-link">AddUMAP()</a></code> in the R package <code>PRECAST</code> to obtain the three-dimensional components of UMAP using the aligned embeddings in the reduction <code>harmony</code>.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seuInt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PRECAST/man/AddUMAP.html" class="external-link">AddUMAP</a></span><span class="op">(</span><span class="va">seuInt</span>, n_comp<span class="op">=</span><span class="fl">3</span>, reduction <span class="op">=</span> <span class="st">'iscmeb'</span>, assay <span class="op">=</span> <span class="st">'ProFAST'</span><span class="op">)</span>
<span class="va">seuInt</span>
<span class="co">#&gt; An object of class Seurat </span>
<span class="co">#&gt; 2000 features across 7312 samples within 1 assay </span>
<span class="co">#&gt; Active assay: ProFAST (2000 features, 0 variable features)</span>
<span class="co">#&gt;  4 dimensional reductions calculated: profast, iscmeb, position, UMAP3</span></code></pre></div>
<p>We plot the spatial tNSE RGB plot to illustrate the performance in extracting features.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p13</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PRECAST/man/SpaPlot.html" class="external-link">SpaPlot</a></span><span class="op">(</span><span class="va">seuInt</span>, batch <span class="op">=</span> <span class="cn">NULL</span>, item <span class="op">=</span> <span class="st">"RGB_UMAP"</span>, point_size <span class="op">=</span> <span class="fl">1</span>, combine <span class="op">=</span> <span class="cn">FALSE</span>, text_size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/PRECAST/man/drawFigs.html" class="external-link">drawFigs</a></span><span class="op">(</span><span class="va">p13</span>, layout.dim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">3</span><span class="op">)</span>, common.legend <span class="op">=</span> <span class="cn">TRUE</span>, legend.position <span class="op">=</span> <span class="st">"right"</span>, align <span class="op">=</span> <span class="st">"hv"</span><span class="op">)</span></code></pre></div>
<p><img src="ProFASTdlpfc2_files/figure-html/unnamed-chunk-22-1.png" width="768"> We use the function <code>AddUTSNE()</code> in the R package <code>PRECAST</code> to obtain the two-dimensional components of UMAP using the aligned embeddings in the reduction <code>iscmeb</code>, and plot the tSNE plot based on the extracted features to check the performance of integration.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seuInt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PRECAST/man/AddTSNE.html" class="external-link">AddTSNE</a></span><span class="op">(</span><span class="va">seuInt</span>, n_comp <span class="op">=</span> <span class="fl">2</span>, reduction <span class="op">=</span> <span class="st">'iscmeb'</span>, assay <span class="op">=</span> <span class="st">'ProFAST'</span><span class="op">)</span>
<span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PRECAST/man/dimPlot.html" class="external-link">dimPlot</a></span><span class="op">(</span><span class="va">seuInt</span>, item <span class="op">=</span> <span class="st">"cluster"</span>, point_size <span class="op">=</span> <span class="fl">0.5</span>, font_family <span class="op">=</span> <span class="st">"serif"</span>, cols <span class="op">=</span> <span class="va">cols_cluster</span>,
    border_col <span class="op">=</span> <span class="st">"gray10"</span>,  legend_pos <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span>  <span class="co"># Times New Roman</span>
<span class="va">p2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PRECAST/man/dimPlot.html" class="external-link">dimPlot</a></span><span class="op">(</span><span class="va">seuInt</span>, item <span class="op">=</span> <span class="st">"batch"</span>, point_size <span class="op">=</span> <span class="fl">0.5</span>, font_family <span class="op">=</span> <span class="st">"serif"</span>, legend_pos <span class="op">=</span> <span class="st">"right"</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/PRECAST/man/drawFigs.html" class="external-link">drawFigs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">p1</span>, <span class="va">p2</span><span class="op">)</span>, layout.dim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>, legend.position <span class="op">=</span> <span class="st">"right"</span>, align <span class="op">=</span> <span class="st">"hv"</span><span class="op">)</span></code></pre></div>
<p><img src="ProFASTdlpfc2_files/figure-html/unnamed-chunk-23-1.png" width="768"></p>
</div>
<div class="section level2">
<h2 id="combined-differential-epxression-analysis">Combined differential epxression analysis<a class="anchor" aria-label="anchor" href="#combined-differential-epxression-analysis"></a>
</h2>
<p>Finally, we condut the combined differential expression analysis using the integrated log-normalized expression matrix saved in the <code>seuInt</code> object. The function <code><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html" class="external-link">FindAllMarkers()</a></code> in the <code>Seurat</code> R package is ued to achieve this analysis.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat_deg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/FindAllMarkers.html" class="external-link">FindAllMarkers</a></span><span class="op">(</span><span class="va">seuInt</span><span class="op">)</span>
<span class="co">#&gt; Calculating cluster 1</span>
<span class="co">#&gt; Calculating cluster 2</span>
<span class="co">#&gt; Calculating cluster 3</span>
<span class="co">#&gt; Calculating cluster 4</span>
<span class="co">#&gt; Calculating cluster 5</span>
<span class="co">#&gt; Calculating cluster 6</span>
<span class="co">#&gt; Calculating cluster 7</span>
<span class="co">#&gt; Calculating cluster 8</span>
<span class="co">#&gt; Calculating cluster 9</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="co">#&gt; Warning: package 'dplyr' was built under R version 4.1.3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following object is masked from 'package:AnnotationDbi':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     select</span>
<span class="co">#&gt; The following objects are masked from 'package:IRanges':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     collapse, desc, intersect, setdiff, slice, union</span>
<span class="co">#&gt; The following objects are masked from 'package:S4Vectors':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     first, intersect, rename, setdiff, setequal, union</span>
<span class="co">#&gt; The following object is masked from 'package:Biobase':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     combine</span>
<span class="co">#&gt; The following objects are masked from 'package:BiocGenerics':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     combine, intersect, setdiff, union</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="va">n</span> <span class="op">&lt;-</span> <span class="fl">5</span>
<span class="va">dat_deg</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html" class="external-link">top_n</a></span><span class="op">(</span>n <span class="op">=</span> <span class="va">n</span>, wt <span class="op">=</span> <span class="va">avg_log2FC</span><span class="op">)</span> <span class="op">-&gt;</span> <span class="va">top10</span></code></pre></div>
<p>Plot dot plot of normalized expressions of the layer-specified marker genes for each spatial domain identified by using the ProFAST embeddings.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">col_here</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#F2E6AB"</span>, <span class="st">"#9C0141"</span><span class="op">)</span> 
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="va">marker_Seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"151507_DEGmarkerTop_5.rds"</span><span class="op">)</span>
<span class="va">marker_Seurat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">marker_Seurat</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="../reference/transferGeneNames.html">transferGeneNames</a></span><span class="op">(</span><span class="va">x</span>, Method<span class="op">=</span><span class="st">'eg.db'</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span>
<span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span>
<span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span>
<span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span>
<span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span>
<span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span>
<span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span>
<span class="va">maker_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">marker_Seurat</span>, <span class="va">toupper</span><span class="op">)</span><span class="op">)</span>
<span class="va">features</span> <span class="op">&lt;-</span> <span class="va">maker_genes</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/duplicated.html" class="external-link">duplicated</a></span><span class="op">(</span><span class="va">maker_genes</span><span class="op">)</span><span class="op">]</span>
<span class="va">p_dot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DotPlot.html" class="external-link">DotPlot</a></span><span class="op">(</span><span class="va">seuInt</span>, features<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/unname.html" class="external-link">unname</a></span><span class="op">(</span><span class="va">features</span><span class="op">)</span>, cols<span class="op">=</span><span class="va">col_here</span>, <span class="co">#  idents = ident_here,</span>
              col.min <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, col.max <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"italic"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Domain"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">12</span>, angle <span class="op">=</span> <span class="fl">25</span>, hjust <span class="op">=</span> <span class="fl">1</span>, family<span class="op">=</span><span class="st">'serif'</span><span class="op">)</span>,
                                      axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">12</span>, face<span class="op">=</span> <span class="st">"italic"</span>, family<span class="op">=</span><span class="st">'serif'</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Warning in FetchData.Seurat(object = object, vars = features, cells = cells):</span>
<span class="co">#&gt; The following requested variables were not found: SNORC</span>
<span class="va">p_dot</span></code></pre></div>
<p><img src="ProFASTdlpfc2_files/figure-html/unnamed-chunk-25-1.png" width="768"></p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># table(unlist(yList), Idents(seuInt))</span></code></pre></div>
<p>Based on the marker genes, we annotate the Domains 1-9 as Layer5/6, Layer3/4, Layer5, Layer2/3, Layer2/3, Layer6, WM, Layer5, Layer1. Then, we rename the Domains using the function <code><a href="https://mojaveazure.github.io/seurat-object/reference/Idents.html" class="external-link">RenameIdents()</a></code>.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">seuInt2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/Idents.html" class="external-link">RenameIdents</a></span><span class="op">(</span><span class="va">seuInt</span>, <span class="st">'1'</span> <span class="op">=</span> <span class="st">'Layer5/6'</span>, <span class="st">'2'</span> <span class="op">=</span> <span class="st">'Layer3/4'</span>, <span class="st">'3'</span><span class="op">=</span><span class="st">"Layer5"</span>, <span class="st">'4'</span><span class="op">=</span><span class="st">"Layer2/3"</span>,
                           <span class="st">'5'</span><span class="op">=</span><span class="st">'Layer2/3'</span>, <span class="st">'6'</span><span class="op">=</span> <span class="st">'Layer6'</span>, <span class="st">'7'</span> <span class="op">=</span><span class="st">'WM'</span>, <span class="st">'8'</span><span class="op">=</span><span class="st">'Layer5'</span>, <span class="st">'9'</span><span class="op">=</span><span class="st">'Layer1'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">seuInt2</span><span class="op">)</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Layer1"</span>, <span class="st">"Layer2/3"</span>, <span class="st">"Layer3/4"</span>, <span class="st">"Layer5"</span>, <span class="st">"Layer5/6"</span>, <span class="st">"Layer6"</span>, <span class="st">"WM"</span><span class="op">)</span>
<span class="va">p_dot2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/DotPlot.html" class="external-link">DotPlot</a></span><span class="op">(</span><span class="va">seuInt2</span>, features<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/unname.html" class="external-link">unname</a></span><span class="op">(</span><span class="va">features</span><span class="op">)</span>, cols<span class="op">=</span><span class="va">col_here</span>, <span class="co">#  idents = ident_here,</span>
              col.min <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, col.max <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html" class="external-link">coord_flip</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face <span class="op">=</span> <span class="st">"italic"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Domain"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">12</span>, angle <span class="op">=</span> <span class="fl">25</span>, hjust <span class="op">=</span> <span class="fl">1</span>, family<span class="op">=</span><span class="st">'serif'</span><span class="op">)</span>,
                                      axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">12</span>, face<span class="op">=</span> <span class="st">"italic"</span>, family<span class="op">=</span><span class="st">'serif'</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; Warning in FetchData.Seurat(object = object, vars = features, cells = cells):</span>
<span class="co">#&gt; The following requested variables were not found: SNORC</span>
<span class="va">p_dot2</span></code></pre></div>
<p><img src="ProFASTdlpfc2_files/figure-html/unnamed-chunk-26-1.png" width="768"></p>
</div>
<div class="section level2">
<h2 id="combined-cell-cell-interaction-analysis">Combined cell-cell interaction analysis<a class="anchor" aria-label="anchor" href="#combined-cell-cell-interaction-analysis"></a>
</h2>
<p>Finally, we demonstrate the combined cell-cell interaction analysis using the R package <code>CellChat</code>.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">CellChat</span><span class="op">)</span><span class="op">)</span>
<span class="va">data.input</span> <span class="op">=</span>  <span class="va">seuInt2</span><span class="op">[[</span><span class="st">"ProFAST"</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">data</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">seuInt2</span><span class="op">[[</span><span class="st">"ProFAST"</span><span class="op">]</span><span class="op">]</span><span class="op">@</span><span class="va">data</span><span class="op">)</span><span class="co"># normalized data matrix with a movement to make the value not smaller than zero.</span>
<span class="va">meta</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>labels<span class="op">=</span><span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/Idents.html" class="external-link">Idents</a></span><span class="op">(</span><span class="va">seuInt2</span><span class="op">)</span><span class="op">)</span> <span class="co"># a dataframe with rownames containing cell mata data</span>
<span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">meta</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">data.input</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">meta</span><span class="op">)</span>
<span class="co"># Prepare input data for CelChat analysis</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">data.input</span><span class="op">)</span>
<span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CellChat/man/createCellChat.html" class="external-link">createCellChat</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">data.input</span>, meta <span class="op">=</span> <span class="va">meta</span>, group.by <span class="op">=</span> <span class="st">"labels"</span><span class="op">)</span>
<span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CellChat/man/addMeta.html" class="external-link">addMeta</a></span><span class="op">(</span><span class="va">cellchat</span>, meta <span class="op">=</span> <span class="va">meta</span><span class="op">)</span>
<span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CellChat/man/setIdent.html" class="external-link">setIdent</a></span><span class="op">(</span><span class="va">cellchat</span>, ident.use <span class="op">=</span> <span class="st">"labels"</span><span class="op">)</span> <span class="co"># set "labels" as default cell identity</span>
<span class="fu"><a href="https://rdrr.io/r/base/levels.html" class="external-link">levels</a></span><span class="op">(</span><span class="va">cellchat</span><span class="op">@</span><span class="va">idents</span><span class="op">)</span> <span class="co"># show factor levels of the cell labels</span>
<span class="va">groupSize</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">cellchat</span><span class="op">@</span><span class="va">idents</span><span class="op">)</span><span class="op">)</span> <span class="co">#</span></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Set the ligand-receptor interaction database</span>
<span class="va">CellChatDB</span> <span class="op">&lt;-</span> <span class="va">CellChatDB.human</span> <span class="co"># use CellChatDB.mouse if running on mouse data</span>
<span class="fu"><a href="https://rdrr.io/pkg/CellChat/man/showDatabaseCategory.html" class="external-link">showDatabaseCategory</a></span><span class="op">(</span><span class="va">CellChatDB</span><span class="op">)</span>
<span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="va">CellChatDB</span><span class="op">$</span><span class="va">interaction</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">cellchat</span><span class="op">@</span><span class="va">DB</span> <span class="op">&lt;-</span> <span class="va">CellChatDB</span>
<span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CellChat/man/subsetData.html" class="external-link">subsetData</a></span><span class="op">(</span><span class="va">cellchat</span><span class="op">)</span> <span class="co"># This step is necessary even if using the whole database</span>
<span class="fu">future</span><span class="fu">::</span><span class="fu"><a href="https://future.futureverse.org/reference/plan.html" class="external-link">plan</a></span><span class="op">(</span><span class="st">"multiprocess"</span>, workers <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="co"># do parallel</span>
<span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CellChat/man/identifyOverExpressedGenes.html" class="external-link">identifyOverExpressedGenes</a></span><span class="op">(</span><span class="va">cellchat</span><span class="op">)</span>
<span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CellChat/man/identifyOverExpressedInteractions.html" class="external-link">identifyOverExpressedInteractions</a></span><span class="op">(</span><span class="va">cellchat</span><span class="op">)</span>
<span class="co"># project gene expression data onto PPI network (optional)</span>
<span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CellChat/man/projectData.html" class="external-link">projectData</a></span><span class="op">(</span><span class="va">cellchat</span>, <span class="va">PPI.human</span><span class="op">)</span>
<span class="co">## Compute the communication probability and infer cellular communication network</span>
<span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CellChat/man/computeCommunProb.html" class="external-link">computeCommunProb</a></span><span class="op">(</span><span class="va">cellchat</span><span class="op">)</span>
<span class="co"># Filter out the cell-cell communication if there are only few number of cells in certain cell groups</span>
<span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CellChat/man/filterCommunication.html" class="external-link">filterCommunication</a></span><span class="op">(</span><span class="va">cellchat</span>, min.cells <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>
<span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CellChat/man/computeCommunProbPathway.html" class="external-link">computeCommunProbPathway</a></span><span class="op">(</span><span class="va">cellchat</span><span class="op">)</span>
<span class="va">cellchat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CellChat/man/aggregateNet.html" class="external-link">aggregateNet</a></span><span class="op">(</span><span class="va">cellchat</span><span class="op">)</span>
<span class="va">groupSize</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">cellchat</span><span class="op">@</span><span class="va">idents</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/CellChat/man/netVisual_circle.html" class="external-link">netVisual_circle</a></span><span class="op">(</span><span class="va">cellchat</span><span class="op">@</span><span class="va">net</span><span class="op">$</span><span class="va">count</span>, vertex.weight <span class="op">=</span> <span class="va">groupSize</span>, weight.scale <span class="op">=</span> <span class="cn">T</span>, label.edge<span class="op">=</span> <span class="cn">F</span>, title.name <span class="op">=</span> <span class="st">"Number of interactions"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/CellChat/man/netVisual_circle.html" class="external-link">netVisual_circle</a></span><span class="op">(</span><span class="va">cellchat</span><span class="op">@</span><span class="va">net</span><span class="op">$</span><span class="va">weight</span>, vertex.weight <span class="op">=</span> <span class="va">groupSize</span>, weight.scale <span class="op">=</span> <span class="cn">T</span>, label.edge<span class="op">=</span> <span class="cn">F</span>, title.name <span class="op">=</span> <span class="st">"Interaction weights/strength"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">het1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CellChat/man/netVisual_heatmap.html" class="external-link">netVisual_heatmap</a></span><span class="op">(</span><span class="va">cellchat</span>, signaling <span class="op">=</span> <span class="cn">NULL</span>,measure <span class="op">=</span> <span class="st">"count"</span>,
                          font.size <span class="op">=</span> <span class="fl">12</span>, font.size.title <span class="op">=</span> <span class="fl">14</span><span class="op">)</span> <span class="co"># color.heatmap = "Reds"</span>


<span class="va">het2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CellChat/man/netVisual_heatmap.html" class="external-link">netVisual_heatmap</a></span><span class="op">(</span><span class="va">cellchat</span>, signaling <span class="op">=</span> <span class="cn">NULL</span>,measure <span class="op">=</span> <span class="st">"weight"</span>, font.size <span class="op">=</span> <span class="fl">12</span>, 
                          font.size.title <span class="op">=</span> <span class="fl">14</span><span class="op">)</span>
<span class="va">het2</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Wei Liu, Xiao Zhang, Jin Liu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
